postcss.config.js

```diff
@@ -0,0 +1,54 @@
+const autoprefixer = require('autoprefixer')
+const cssnano = require('cssnano')
+const oldie = require('oldie')
+const pseudoclasses = require('postcss-pseudo-classes')
+
+/**
+ * PostCSS config
+ *
+ * @type {import('postcss-load-config').ConfigFn}
+ */
+module.exports = ({ env, file }) => {
+  const plugins = [
+    autoprefixer({ env })
+  ]
+
+  // Minify CSS
+  if (file.basename.endsWith('.min.css')) {
+    plugins.push(cssnano())
+  }
+
+  // Auto-generate 'companion' classes for pseudo-selector states - e.g. a
+  // :hover class you can use to simulate the hover state in the review app
+  if (file.basename.startsWith('app-')) {
```

Comment:
The original logic for this looks like it was "[anything that's not the `dist` folder](https://github.com/alphagov/govuk-frontend/blob/023c3bfa60470c9b614d0e34eaf23a2eba119529/tasks/gulp/compile-assets.js#L60)".

I get the other two `if` conditions, but for some reason I'm struggling to figure this change out, so I've probably missed something obvious - could you explain it for me?

Reply:
Ah, no, I should've read the surrounding code - so this stuff is applied to `app/assets/scss/app.scss` in `compileStyles` `app/assets/scss/app-legacy.scss` in `compileLegacy` and `app/assets/scss/app-legacy-ie8.scss` in `compileLegacyIE`.

So my revised query is why `app.scss` isn't included here?

Reply:
I've pushed again due to a slight complication

1. Plugin for Gulp `gulp-postcss` populates PostCSS **ctx.file** as type [`import('vinyl')`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/vinyl/index.d.ts#L75)
2. Plugin for webpack `postcss-loader` populates PostCSS **ctx.file** as a file path `string`

Found this out using the webpack example https://github.com/alphagov/govuk-frontend/pull/2869

Additionally the Gulp stream `ctx.file.basename` will be **.css** but other tools pass the original **.scss**

Take a look at the new version 👍 

Thanks @domoscargin